src/Telegram.Bot.Extensions.Exceptions/ExceptionParser.cs

```diff
@@ -1,17 +1,150 @@
+                "invalid sticker emojis"
+            ),
+            new BadRequestExceptionDescriptor<InvalidStickerDimensionsException>(
+                "STICKER_PNG_DIMENSIONS"
+            ),
+            new BadRequestExceptionDescriptor<StickerSetNameExistsException>(
+                "sticker set name is already occupied"
+            ),
+            new BadRequestExceptionDescriptor<StickerSetNotModifiedException>(
+                "STICKERSET_NOT_MODIFIED"
+            ),
+
+            #endregion
+
+            #region Games
+
+            new BadRequestExceptionDescriptor<InvalidGameShortNameException>(
+                "GAME_SHORTNAME_INVALID"
+            ),
+            new BadRequestExceptionDescriptor<InvalidGameShortNameException>(
+                "game_short_name is empty"
+            ),
+            new BadRequestExceptionDescriptor<InvalidGameShortNameException>(
+                "wrong game short name specified"
+            ),
+
+            #endregion
+
+            new BadRequestExceptionDescriptor<ContactRequestException>(
+                "phone number can be requested in private chats only"
+            ),
+
+            new ForbiddenExceptionDescriptor<ChatNotInitiatedException>(
+                "bot can't initiate conversation with a user"
+            ),
+
+            new BadRequestExceptionDescriptor<InvalidParameterException>(
+                $@"Bad Request: invalid (?<{InvalidParameterException.ParamGroupName}>[\w|\s]+)$"
+            )
+            {
+                CustomExceptionFactory = InvalidParameterExceptionFactory
+            },
+            new BadRequestExceptionDescriptor<InvalidParameterException>(
+                $@"Bad Request: (?<{InvalidParameterException.ParamGroupName}>[\w|\s]+) invalid$"
+            )
+            {
+                CustomExceptionFactory = InvalidParameterExceptionFactory
+            },
+
+            new BadRequestExceptionDescriptor<MessageIsNotModifiedException>(
+                "message is not modified"
+            ),
+        };
+
+        private readonly IApiExceptionDescriptor[] _exceptionDescriptors;
+
+        internal ExceptionParser(IEnumerable<IApiExceptionDescriptor> exceptionDescriptors)
+        {
+            if (exceptionDescriptors is null)
+                throw new ArgumentNullException(nameof(exceptionDescriptors));
+
+            _exceptionDescriptors = exceptionDescriptors.ToArray();
+        }
+
+        public ApiRequestException Parse(
+            int errorCode,
+            string description,
+            ResponseParameters? responseParameters)
+        {
+            ApiRequestException? exception;
+
+            foreach (var descriptor in _exceptionDescriptors)
+            {
+                if (descriptor.TryParseException(
+                    errorCode,
+                    description,
+                    responseParameters,
+                    out exception))
+                {
+                    return exception;
```

Comment:
```suggestion
                    return exception ?? throw new ArgumentNullException(nameof(exception), "Parsed exception must not be null if TryParseException returned true or something along those lines)";
```

I think we should enforce the nullability contract here - idk if `Argument*` exception is the right fit here tho

Reply:
It's not an argument exception, so it should be something else I think. It's violation of a contract, maybe `InvalidOperationException` then?

Reply:
I'm okay with `InvalidOperationException`, since the value would be in the exception message anyway